lib/api/v1/forms/upload_keywords.go

```diff
@@ -0,0 +1,80 @@
+package forms
+
+import (
+	"encoding/csv"
+	"errors"
+	"io"
+	"mime/multipart"
+
+	"github.com/markgravity/golang-ic/database"
+	"github.com/markgravity/golang-ic/lib/jobs"
+	"github.com/markgravity/golang-ic/lib/models"
+
+	"github.com/fatih/structs"
+)
+
+type UploadKeywordsForm struct {
+	FileHeader *multipart.FileHeader `form:"file" binding:"required"`
+	User       *models.User
+}
+
+func (f *UploadKeywordsForm) Save() error {
+	db := database.GetDB()
+	keywords, err := f.readCSVFile()
+	if err != nil {
+		return err
+	}
+
+	for _, k := range keywords {
+		keyword := &models.Keyword{
+			Text: k,
+			User: f.User,
+		}
+
+		err := keyword.Save(db)
+		if err != nil {
+			return err
+		}
+
+		job := jobs.Crawl{}
+		job.SetArgs(structs.Map(keyword))
+		err = jobs.Dispatch(&job)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (f *UploadKeywordsForm) readCSVFile() ([]string, error) {
```

Comment:
```suggestion
func (f *UploadKeywordsForm) readKeywordsFromCSVFile() ([]string, error) {
```

Read CSV is a bit implicit about what the function is supposed to return (the header? the meta-data, the content in a single string? the CSV rows? ...). If that function returns keywords, let's tell it! :)

(but if the form is named `KeywordsCSV`, we can simplify with `readKeywords` maybe? :)